app/services/setup_slug.rb

```diff
@@ -0,0 +1,71 @@
+# if we have class - it's public behaviour should be tested
+# try and minimize all classes public behaviour
+# dont' directly private behaviour
+
+# SOLID
+
+class SetupSlug
+
+  def self.run(name)
+    return if name.nil?
+    new(name).send(:candidates)
+  end
+
+  private
+
+  attr_reader :name
+
+  def initialize(name)
+    @name = name
+  end
+
+  def candidates
+    [  name.method(:short_name),
+        name.method(:longer_name),
+        name.method(:append_org),
+       :name
+    ]
+  end
+
+end
+
+# unit test should be added for the following methods
+# string_spec.rb
+
+class String
```

Comment:
so we were going to break this out into a separate file, e.g. lib/String.rb, and add an associated test? i..e string_spec.rb - work of integration test could be pulled into string_spec.rb ?